#! /bin/sh /usr/share/dpatch/dpatch-run
## 100327_r24850_bootstraptest_test_thread.dpatch by Daigo Moriwaki <daigo@debian.org>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Back port from the upstream r24850, fixing a test failure.

@DPATCH@
Index: b/bootstraptest/test_thread.rb
===================================================================
--- a/bootstraptest/test_thread.rb	2010-07-20 21:45:44.000000000 +0900
+++ b/bootstraptest/test_thread.rb	2010-07-20 21:46:18.000000000 +0900
@@ -188,18 +188,18 @@
     Thread.current[:a]
   }.value + Thread.current[:a]
 }
-assert_equal %q{100}, %q{
+assert_normal_exit %q{
 begin
   100.times do |i|
     begin
-      Thread.start(Thread.current) {|u| u.raise }
+      th = Thread.start(Thread.current) {|u| u.raise }
       raise
     rescue
     ensure
+      th.join
     end
   end
 rescue
-  100
 end
 }, '[ruby-dev:31371]'
 
